staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -322,49 +328,30 @@ func TestApfCancelWaitRequest(t *testing.T) {
 		t.Error(err)
 	}
 
-	checkForExpectedMetricsWithRetry(t, []string{
+	checkForExpectedMetrics(t, []string{
 		"apiserver_current_inflight_requests",
 		"apiserver_request_terminations_total",
 		"apiserver_dropped_requests_total",
 	})
 }
 
 // wait async metrics to be collected
-func checkForExpectedMetricsWithRetry(t *testing.T, expectedMetrics []string) {
-	maxRetries := 5
-	var checkErrors []error
-	for i := 0; i < maxRetries; i++ {
```

Comment:
What changed to make these retries unnecessary?

Reply:
from https://github.com/kubernetes/kubernetes/pull/96244#pullrequestreview-523914035:

> This PR contains the changes in #96229 and adds a simplification, removing retries that existed only as an attempt to cope with one of the races.

Reply:
The retries were necessary only because the test funcs proceeded to look for metrics without waiting for the request processing to finish.  That is the data race that has been fixed.  So the retries are no longer necessary.